File: Entities/Concrete/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Core.Entities;

namespace Entities.Concrete
{
    public class Product : IEntity
    {
        public int Id { get; set; }
        public int CategoryId { get; set; } //İlişki anahtarı Foreing Key
        public string ProductName { get; set; }
        public decimal Price { get; set; }
        public int UnitsInStock { get; set; }
        public string Description { get; set; }
        public string ImageUrl { get; set; }
        public bool IsActive { get; set; } = true;
    }
}

File: UI/src/app/models/product.ts
export interface Product {
    id: number;
    categoryId: number;
    productName: string;
    price: number;        // Backend: Price (decimal)
    unitsInStock: number; // Backend: UnitsInStock
    description: string;  // Backend: Description
    imageUrl: string;     // Backend: ImageUrl
    isActive: boolean;
}

// Backend'den gelen genel yanıt kalıbı (Result)
export interface DataResult<T> {
    success: boolean;
    message: string;
    data: T;
}
